framework/src/modules/chain/blocks/chain.js

```diff
@@ -94,18 +93,30 @@ const saveBlock = async (storage, block, tx) => {
 };
 
 /**
- * Deletes block from blocks table.
+ * Deletes last block.
  *
- * @param {number} blockId - ID of block to delete
- * @param {function} cb - Callback function
- * @param {Object} tx - Database transaction
+ * @param  {function} cb - Callback function
  * @returns {function} cb - Callback function from params (through setImmediate)
- * @returns {Object} cb.err - String if SQL error occurred, null if success
+ * @returns {Object} cb.err - Error if occurred
+ * @returns {Object} cb.obj - New last block
  */
-const deleteBlock = async (storage, blockId, tx) =>
-	// Delete block with ID from blocks table
-	// WARNING: DB_WRITE
-	storage.entities.Block.delete({ id: blockId }, {}, tx);
+const deleteLastBlock = async (storage, lastBlock, tx) => {
+	if (lastBlock.height === 1) {
+		throw new Error('Cannot delete genesis block');
+	}
+	const [storageBlock] = await this.storage.entities.Block.get(
+		{ id: lastBlock.previousBlock },
+		{ extended: true },
```

Comment:
Is it necessary to request the extended block? We are just checking if the block exists and using the `id`

Reply:
it's cacheing the block as last block so it should be extended